From 6d1f1d298381ca14d3c6c6708b1c8b9f448dd63c Mon Sep 17 00:00:00 2001
From: Arne Coucheron <arco68@gmail.com>
Date: Sun, 1 Dec 2019 04:52:44 +0100
Subject: [PATCH] [DNM] libdl: hack: Make legacy camera great again

The real issue is caused by

4e46ac "Load libc_malloc_* libraries from the runtime APEX"

which includes a static version of libdl_android in libc.
For some reason this causes an ILL_OPC from mm-qcamera-daemon.

Another issue with legacy camera (not related to this patch though),
is that Google removed /dev/msm_camera from their system ueventd.rc.

See https://review.lineageos.org/c/LineageOS/android_device_samsung_jf-common/+/265354

Change-Id: Ibede7b9360d4d4f3b43ab22184e4c743b8bfbfed
---
 libdl/libdl_android.cpp | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/libdl/libdl_android.cpp b/libdl/libdl_android.cpp
index 9ad8250..a4e2d4c 100644
--- a/libdl/libdl_android.cpp
+++ b/libdl/libdl_android.cpp
@@ -123,7 +123,8 @@ struct android_namespace_t* android_get_exported_namespace(const char* name) {
 // would create a dependency cycle with libc.so.
 __attribute__((visibility("hidden")))
 void __aeabi_unwind_cpp_pr0() {
-  __builtin_trap();
+//  __builtin_trap();
+  void();
 }
 #endif
 
-- 
2.7.4

